---
Q5. For the city of Bengaluru, how many days fell under each air quality category 
(e.g., Good, Moderate, Poor, etc.) between March and May 2025?
---

In [10]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [11]:
data=pd.read_csv('day-wise-state-wise-air-quality-index-aqi-of-major-cities-and-towns-in-india.csv')

In [12]:
# Filtering data for Bengaluru city between March and May 2025
data['date'] = pd.to_datetime(data['date'],format='%d-%m-%Y') # add this before filtering
Bengaluru_data=data[(data['area']=='Bengaluru') & (data['date']>='2025-03-01') & (data['date']<='2025-05-31')].copy()
result=Bengaluru_data['air_quality_status'].value_counts()
result

air_quality_status
Satisfactory    74
Moderate        13
Good             5
Name: count, dtype: int64

In [13]:
# Further analysis of Bangalore's mean AQI during this period
avg_aqi=Bengaluru_data['aqi_value'].mean()
avg_aqi

78.80434782608695

In [14]:
#overall avg aqi of Bengaluru 
df=data[data['area']=='Bengaluru']
df['aqi_value'].mean()

75.51559080962801

### Key Insights
- The average AQI for Bengaluru(~ 78.80) is below the critical value and is fairly moderate most of the days. Targeted campaigns for health-conscious customers should be implemented.
- **Bengaluru is not a crisis market**- It is a health consciousness market. With an average AQI of 78 and zero severe days, consumers here will not buy an air purifier out of fear or necessity. They will buy it as a lifestyle and wellness investment.